src/main/java/de/unijena/cheminf/fragment/fingerprint/CountFingerprint.java

```diff
@@ -0,0 +1,179 @@
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package de.unijena.cheminf.fragment.fingerprint;
+
+import org.openscience.cdk.fingerprint.ICountFingerprint;
+
+import java.util.HashMap;
+
+/**
+ * The CountFingerprint class implements the CDK interface ICountFingerprint.
+ * ICountFingerprint provides useful methods to obtain information about the calculated count fingerprint.
+ */
+public class CountFingerprint implements ICountFingerprint {
+    //<editor-fold desc="private  class variables" defaultstate="collapsed">
+    /**
+     * Is an array containing all predefined unique SMILES.
+     */
+    private String[] fragmentArrayOfUniqueSmiles;
+    /**
+     * The HashMap maps the unique SMILES to the position they have in the array.
+     */
+    private HashMap<String,Integer> uniqueSmilesToPositionMap;
+    /**
+     * Map
+     */
+    private HashMap<Integer, Integer> rawMap;
+    //</editor-fold>
+    //
+    //<editor-fold desc="Constructor" defaultstate="collapsed">
+    /**
+     * Constructor.
+     *
+     * @param aFragments
+     * @param aMapOfFragmentSmiles
+     */
+    public CountFingerprint(String[] aFragments, HashMap<String, Integer> aMapOfFragmentSmiles, HashMap<Integer, Integer> aRawMap) {
+        this.fragmentArrayOfUniqueSmiles = aFragments;
+        this.uniqueSmilesToPositionMap = aMapOfFragmentSmiles;
+        this.rawMap = aRawMap;
+    }
+    //</editor-fold>
+    //
+    //<editor-fold desc="Overridden public methods " defaultstate="collapsed">
+    /**
+     * Returns the number of bits in this fingerprint.
+     * Since fragment fingerprints are key based, the number of bits in the fingerprint
+     * is equal to the number of predefined fragments (unique SMILES).
+     *
+     * @return the size of the fingerprint.
+     */
+    @Override
+    public long size() {
+        return this.fragmentArrayOfUniqueSmiles.length;
+    }
+    //
+    /**
+     * Returns the number of bins that are populated.
+     * The number is equal to the defined fragments.
+     *
+     * @return the number of populated bins
+     * @see    #size()
+     */
+    @Override
+    public int numOfPopulatedbins() {
+        return this.fragmentArrayOfUniqueSmiles.length;
+    }
+    //
+    /**
+     * For each defined structure in the fingerprint, returns the frequency.
```

Comment:
Please change to a "normal" sentence structure. And specify "structure" as e.g. fingerprint position index.